src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -313,12 +321,19 @@ private IDisposable RegisterForNotificationOnNewDocuments(SubscriptionCriteria c
             using (DisposeOnDisconnect)
             using (TcpConnection.DocumentDatabase.DocumentsStorage.ContextPool.AllocateOperationContext(out dbContext))
             {
-                long startEtag;
+                ChangeVectorEntry[] startChangeVector;
                 SubscriptionCriteria criteria;
 
-
                 TcpConnection.DocumentDatabase.SubscriptionStorage.GetCriteriaAndEtag(_options.SubscriptionId, dbContext,
-                    out criteria, out startEtag);
+                    out criteria, out startChangeVector);                
+                
+                long startEtag = 0;
+                var dbId = TcpConnection.DocumentDatabase.DbId;
+                foreach (var changeVectorEntry in startChangeVector)
```

Comment:
This isn't how we can handle this. We need to actually do a search on the etags collection based on the change vector.